encoding/dachunk.go

```diff
@@ -0,0 +1,192 @@
+package encoding
+
+import (
+	"encoding/binary"
+	"encoding/hex"
+	"errors"
+	"fmt"
+	"strings"
+
+	"github.com/scroll-tech/go-ethereum/common"
+	"github.com/scroll-tech/go-ethereum/core/types"
+	"github.com/scroll-tech/go-ethereum/crypto"
+)
+
+// DAChunkV0 groups consecutive DABlocks with their transactions.
+type DAChunkV0 struct {
+	blocks       []DABlock
+	transactions [][]*types.TransactionData
+}
+
+// NewDAChunkV0 is a constructor for DAChunkV0, initializing with blocks and transactions.
+func NewDAChunkV0(blocks []DABlock, transactions [][]*types.TransactionData) *DAChunkV0 {
```

Comment:
These constructor methods (same for `NewDABlockVX` and `NewDABatchVX`) should be only used internally by the respective codec. We should try to keep the outside exposed interface of the entire package as small as possible to make it easier to use. Since we're returning interfaces, maybe even the types can be private.

Reply:
changed types of specific implementations to internal in [233b006](https://github.com/scroll-tech/da-codec/pull/25/commits/233b006ae0389dd1e3438434e9b2b5c100ae97f0).